### ISSUE TITLE
Error related to JDK in macOS.

### ISSUE THREAD

**Issue by ikusorae** [renpy/renpy#2622]
====================

Ren'Py 7.4.2.1273

I'm compiling a short test program, to see if you have a working JDK on your system.

Traceback (most recent call last):
  File "game/mobilebuild.rpy", line 195, in call
  File "/home/tom/ab/nightly-build/tmp/install.linux-x86_64/lib/python2.7/subprocess.py", line 325, in __init__
  File "/home/tom/ab/nightly-build/tmp/install.linux-x86_64/lib/python2.7/subprocess.py", line 978, in _execute_child
OSError: [Errno 2] No such file or directory

I was unable to use javac to compile a test file. If you haven't installed the Java Development Kit yet, please download it from:

http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html

The JDK is different from the JRE, so it's possible you have Java without having the JDK. Without a working JDK, I can't continue.

====================

Error in "Install SDK & Create Keys" step when attempting to distribute to Android.

JDK8 is already installed in my macOS.
The results were the same in clean mac.

(macOS 11.1)

**Comment by jackmcbarn** [comment]
Create the following file called `Hello.java`:
```java
public class Hello {
    public static void main(String[] args) {
        System.out.println("Hello, World!"); 
    }
}
```
Then open the Terminal and run `javac Hello.java && java Hello`. Does it succeed?

**Comment by ikusorae** [comment]
<img width="563" alt="스크린샷 2021-01-26 오전 3 44 50 사본" src="https://user-images.githubusercontent.com/60040734/105751219-0f7de800-5f89-11eb-85bd-98a56570ef82.png">

Yes. Unlike Ren'py, it works well.

**Comment by renpytom** [comment]
Could you take a look at two things for me? The first is the output of:

```
/usr/libexec/java_home -v 1.8
```

in the terminal. The second would be to go into the Ren'Py launcher, hit shift+O to access the console, and then type:

```
> import os
> os.environ.get("JAVA_HOME", "<not set>")
```

**Comment by ikusorae** [comment]
@renpytom
First:
<img width="506" alt="스크린샷 2021-01-26 오후 4 50 11 사본" src="https://user-images.githubusercontent.com/60040734/105815800-ce242180-5ff6-11eb-937a-5da527e8e6ff.png">

Second:
<img width="379" alt="스크린샷 2021-01-26 오후 4 49 39" src="https://user-images.githubusercontent.com/60040734/105815579-7c7b9700-5ff6-11eb-810c-4ae6b6d1247c.png">

Thanks.

**Comment by renpytom** [comment]
I've seen this before, and it's because for some reason the old plug-in version of Java is taking preference over what you've installed. 

The right way to solve this is to remove all Java from your system, and download the version from: https://adoptopenjdk.net/releases.html?variant=openjdk8&jvmVariant=hotspot 

I'll also look at another way of solving this, but this will get you going.

**Comment by ikusorae** [comment]
I tried the method you told me, and it seems to have been solved. Thank you.

**Comment by renpytom** [comment]
This will be fixed for good by renpy/renpy-build@d20eee9.
